### PR TITLE
Switch to Github Actions for CI

### PR DESCRIPTION
**Pull request by tiennou**
This moves all our CI infrastructure over to Github Actions and removes Travis from the equation, with a matrix of Ubuntu/macOS vs. Ruby 2.4/2.5/2.6/head.

Note that I cherrypicked one commit from #56 to handle Ruby 3/`head`, and that I've run the actions while pushing to my fork's `master` branch — see [here](https://github.com/tiennou/docurium/actions) for the runs.

Loosely inspired from #47 (had an issue with finding libclang on Linux).

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [head, 2.6, 2.5, 2.4]
+        os: [ ubuntu-18.04, macos-latest ]
+
+    name: Ruby ${{ matrix.ruby }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ endsWith(matrix.ruby, 'head') || matrix.ruby == 'debug' }}
```
Comment (carlosmn) [inline .github/workflows/main.yml:20]: Is "debug" left over from some other testing? Is the `endsWith` necessary? Wouldn't `matrix.ruby` match `head` exactly?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Linux packages
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt update
+          sudo apt install -y libclang-6.0-dev llvm-6.0 clang-6.0
```
Comment (carlosmn) [inline .github/workflows/main.yml:28]: This is fine for now but we should also have some tests run with a different version of LLVM. This was only really set to 6 due to some bugs that meant docurium was often broken with different versions.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [head, 2.6, 2.5, 2.4]
```
Comment (carlosmn) [inline .github/workflows/main.yml:15]: While we're updating things, this should probably be
```suggestion
        ruby: [head, 3.0, 2.7, 2.6]
```

as that's the supported versions upstream

File: .github/workflows/main.yml (hunk 1 above)
Comment (tiennou) [reply .github/workflows/main.yml:20]: That's a verbatim copy-paste from the documentation, where there's much more than MRI Ruby handled. I'm tempted to keep it as-is for future-proofing, but I'll gladly defer to your preference.

File: .github/workflows/main.yml (hunk 2 above)
Comment (tiennou) [reply .github/workflows/main.yml:28]: I had a hunch compatibility was the issue. I'll investigate how easy it is to move LLVM version as a matrix parameter.

File: .github/workflows/main.yml (hunk 3 above)
Comment (tiennou) [reply .github/workflows/main.yml:15]: 👍

**Comment by carlosmn** [comment]
It looks like Ruby 2.6 sometimes fails to grab all the groups, but somewhat inconsistently. I wonder what's up with that and whether it's something we need to/can fix.

**Comment by tiennou** [comment]
Yeah, I suspect it's spotty because of stuff like `/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/HEAD-20210603-2774-1tb1dgk/types.h:36:1: error: unknown type name 'GIT_BEGIN_DECL'`, causing whatever "group" header was currently being processed to be entirely missing. I still haven't pinpointed if it's a clang bug that causes it to read an include as a 0-length file, or a clobbering issue on our side, but I know I've seen utterly nonsensical stuff, like `unknown type name 'ypere'`.

I have a patch that implements a retry loop around that to alleviate this issue, and I might upstream it separately.

**Pull request merged**